plugins/email/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/email/EMailLogic.java

```
@@ -62,299 +57,228 @@
-			this.logger.error("error reading message", e);
 		}
-		return sb.toString();
+		return "Message content not readable";
+
 	}
 
 	/**
-	 * sends a message to some recipient
+	 * Get all mails in the inbox and convert them to MessageDTO objects for the REST class to send to the web app
 	 * 
-	 * @param recipient
-	 *            the recipient
-	 * @param subject
-	 *            mail subject
-	 * @param message
-	 *            the mail body
-	 * @return success string
+	 * @param amount
+	 *            the amount of mails, put -1 here if you want all mails
+	 * 
+	 * @return all mails in the inbox in a list, because lists are better to work with than arrays
 	 */
-	public String sendMail(String recipient, String subject, String message) {
-		Message msg = new MimeMessage(this.session);
-
-		InternetAddress addressTo;
+	public List<MessageDTO> getMailsForREST(int amount) {
+		List<MessageDTO> messagesToSend = new ArrayList<>();
+		Message[] messages;
 		try {
-			addressTo = new InternetAddress(recipient);
-			msg.setRecipient(Message.RecipientType.TO, addressTo);
-
-			msg.setSubject(subject);
-			msg.setContent(message, "text/plain");
-			Transport.send(msg);
-		} catch (MessagingException e) {
-			this.logger.error("messaging exception while sending mail", e);
-			return "Message could not be sent";
+			messages = this.mailSession.getInbox().getMessages();
+			int amountToPrint = (amount == -1) ? messages.length : amount;
+			for (int i = 0; i < amountToPrint; i++) {
+				// we get the messages from the inbox from oldest to newest, but we want to send the most recent ones
+				Message m = messages[messages.length - 1 - i];
+				messagesToSend.add(new MessageDTO(getFrom(m), m.getSubject(), getContentFromMessage(m), m.getSentDate(),
+						isImportantMessage(m), m.isSet(Flags.Flag.SEEN)));
+			}
+			return messagesToSend;
+		} catch (MessagingException | IOException e) {
+			this.logger.error("Something went wrong", e);
+			return new ArrayList<>();
```

maybe throw a exception to indicate something went wrong